backend/jobs/jiraJob.ts

```diff
@@ -305,82 +304,95 @@ export const fetchAndSaveJiraData = async () => {
         },
       });
 
-      console.log(`Access token refreshed for course with cloudId: ${cloudId}`);
+      console.log(
+        `Access token refreshed for course with courseId: ${course._id}`
+      );
     } else {
       console.error(
-        `Failed to refresh access token for course with cloudId: ${cloudId}`
+        `Failed to refresh access token for course with courseId: ${course._id}`
       );
     }
 
-    fetch(jiraBoardUri, {
-      method: 'GET',
-      headers: {
-        Authorization: `Bearer ${accessToken}`,
-        Accept: 'application/json',
-        'Content-Type': 'application/json',
-      },
-    })
-      .then(response => {
-        if (!response.ok) {
-          throw new Error('Network response was not ok');
-        }
-        return response.json();
-      })
-      .then(async data => {
-        const boards = data.values;
-
-        boards.forEach(async (boardData: any) => {
-          const boardId = await boardData.id;
-          const jiraBoardConfigurationUri = `https://api.atlassian.com/ex/jira/${cloudId}/rest/agile/1.0/board/${boardId}/configuration`;
-
-          const boardConfigurationResponse = await fetch(
-            jiraBoardConfigurationUri,
-            {
-              method: 'GET',
-              headers: {
-                Authorization: `Bearer ${accessToken}`,
-                Accept: 'application/json',
-              },
-            }
-          );
+    for (const cloudId of cloudIds) {
+      const jiraBoardUri = `https://api.atlassian.com/ex/jira/${cloudId}/rest/agile/1.0/board`;
 
-          if (!boardConfigurationResponse.ok) {
-            throw new Error(
-              `Failed to fetch board configuration. Status: ${boardConfigurationResponse.status}`
-            );
+      fetch(jiraBoardUri, {
```

Comment:
Mixing `async/await` and promises? Maybe we should add to our developer guide to prefer `async/await` since that's the style we've mostly used so far

Reply:
I will make the change from fetch-then to async await later. There is also another file that uses fetch-then that also needs to be changed. Thank you for highlighting it.